api/src/main/java/com/epam/pipeline/manager/pipeline/PipelineRunManager.java

```diff
@@ -825,6 +827,17 @@ public PipelineRun createPipelineRun(String version, PipelineConfiguration confi
         return run;
     }
 
+    private boolean checkRunForSensitivity(Map<String, PipeConfValueVO> parameters) {
+        return parameters.entrySet().stream()
+                .filter(v->v.getKey().equals(CP_CAP_LIMIT_MOUNTS))
+                .map(Map.Entry::getValue)
+                .flatMap(pipeConfValueVO -> Arrays.stream(
+                        StringUtils.commaDelimitedListToStringArray(pipeConfValueVO.getValue()))
+                )
+                .map(Long::valueOf)
+                .anyMatch(id -> dataStorageManager.load(id).isSensitive());
```

Comment:
It would be helpful to add a method to load  a list of storages by id, to reduce number of db requests.